include/openssl/service_indicator.h

```diff
@@ -8,31 +8,26 @@
 extern "C" {
 #endif
 
-struct fips_service_indicator_state {
-  int counter;
-  int serviceID;
-};
-
 OPENSSL_EXPORT int awslc_fips_service_indicator_get_counter(void);
-OPENSSL_EXPORT struct fips_service_indicator_state *awslc_fips_service_indicator_get_state(void);
-OPENSSL_EXPORT void awslc_fips_service_indicator_reset_state(void);
-OPENSSL_EXPORT int awslc_fips_check_service_approved(int prev_counter, int service_id);
-
+OPENSSL_EXPORT int awslc_fips_service_indicator_get_serviceID(void);
+OPENSSL_EXPORT int awslc_fips_service_indicator_reset_state(void);
+
+#if defined(AWSLC_FIPS)
+#define IS_FIPS_APPROVED_CALL_SERVICE(approved, func, ...)                  \
+  do {                                                                      \
+    approved = 0;                                                           \
+    int counter = awslc_fips_service_indicator_get_counter();               \
+    func(__VA_ARGS__);                                                      \
+    if (awslc_fips_service_indicator_get_counter() > counter) {             \
+        approved = 1;                                                       \
+    }                                                                       \
+ }                                                                          \
+ while(0);                                                                  \
```

Comment:
The `do-while{0}` is used to put the code in a scope and to leave off the last `;` so that you can append it to the macro calls as if it were a function call, so I suggest you make this change, Samuel.